### ISSUE TITLE
✅ 🔗 🐞 🐘 Elementor checkout widget throws warning about parsing null parameter to $string in PHP 8.2+ (Duplicate of #24880)

### ISSUE THREAD

**Issue by mondalaci** [elementor/elementor#25392]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

Upon visiting the  checkout page, the following deprecation warning is triggered:

```
E_DEPRECATED: parse_str(): Passing null to parameter #1 ($string) of type string is deprecated
(/var/www/ultimatehackingkeyboard.com/wordpress/wp-content/plugins/elementor-pro/modules/woocommerce/module.php:627)

parse_str(/var/www/ultimatehackingkeyboard.com/wordpress/wp-content/plugins/elementor-pro/modules/woocommerce/module.php:627)

ElementorPro\Modules\Woocommerce\Module::load_widget_before_wc_ajax(/var/www/ultimatehackingkeyboard.com/wordpress/wp-includes/class-wp-hook.php:324)

WP_Hook::apply_filters(/var/www/ultimatehackingkeyboard.com/wordpress/wp-includes/class-wp-hook.php:348)

WP_Hook::do_action(/var/www/ultimatehackingkeyboard.com/wordpress/wp-includes/plugin.php:517)

do_action(/var/www/ultimatehackingkeyboard.com/wordpress/wp-content/plugins/woocommerce/includes/class-wc-cart.php:1406)

WC_Cart::calculate_totals(/var/www/ultimatehackingkeyboard.com/wordpress/wp-content/plugins/woocommerce/includes/class-wc-ajax.php:401)

WC_AJAX::update_order_review(/var/www/ultimatehackingkeyboard.com/wordpress/wp-includes/class-wp-hook.php:324)

WP_Hook::apply_filters(/var/www/ultimatehackingkeyboard.com/wordpress/wp-includes/class-wp-hook.php:348)

WP_Hook::do_action(/var/www/ultimatehackingkeyboard.com/wordpress/wp-includes/plugin.php:517)

do_action(/var/www/ultimatehackingkeyboard.com/wordpress/wp-content/plugins/woocommerce/includes/class-wc-ajax.php:96)

WC_AJAX::do_wc_ajax(/var/www/ultimatehackingkeyboard.com/wordpress/wp-includes/class-wp-hook.php:324)

WP_Hook::apply_filters(/var/www/ultimatehackingkeyboard.com/wordpress/wp-includes/class-wp-hook.php:348)

WP_Hook::do_action(/var/www/ultimatehackingkeyboard.com/wordpress/wp-includes/plugin.php:517)

do_action(/var/www/ultimatehackingkeyboard.com/wordpress/wp-includes/template-loader.php:13)

require_once(/var/www/ultimatehackingkeyboard.com/wordpress/wp-blog-header.php:19)

require(/var/www/ultimatehackingkeyboard.com/wordpress/index.php:17)
```

We use Rollbar for logging PHP errors and warnings, and this is a high-occurrence warning. It adds to the noise of errors and counts toward our Rollbar quota, so it'd be nice to see it resolved.

### Steps to reproduce

1. Add a product to the cart
2. Visit the checkout page

### Expected behavior

No deprecation warnings should be triggered.

### Isolating the problem

- [ ] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [ ] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt

== Server Environment ==
	Operating System: Linux
	Software: nginx/1.24.0
	MySQL version: (Ubuntu) v8.0.36
	PHP Version: 8.2.10-2ubuntu1
	PHP Memory Limit: 256M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 8M
	GD Installed: Yes
	ZIP Installed: No
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.4.3
	Site URL: https://ultimatehackingkeyboard.com/wordpress
	Home URL: https://ultimatehackingkeyboard.com
	WP Multisite: No
	Max Upload Size: 2 MB
	Memory limit: 256M
	Max Memory limit: 256M
	Permalink Structure: /blog/%year%/%monthnum%/%day%/%postname%
	Language: en-US
	Timezone: Europe/Budapest
	Debug Mode: Inactive

== Theme ==
	Name: Avada-UHK
	Version: 1.0.0
	Author: ThemeFusion
	Child Theme: Yes
	Parent Theme Name: Avada
	Parent Theme Version: 7.11.6
	Parent Theme Author: ThemeFusion

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36

== Active Plugins ==
	Aelia Currency Switcher for WooCommerce
		Version: 5.1.3.240205
		Author: Aelia

	Aelia Foundation Classes for WooCommerce
		Version: 2.4.24.240214
		Author: Aelia

	Aelia Tax Display by Country for WooCommerce
		Version: 1.20.11.240214
		Author: Aelia

	Avada Builder
		Version: 3.11.6
		Author: ThemeFusion

	Avada Core
		Version: 5.11.6
		Author: ThemeFusion

	Back In Stock Notifier for WooCommerce | WooCommerce Waitlist Pro
		Version: 5.2.2
		Author: codewoogeek

	Elementor
		Version: 3.19.4
		Author: Elementor.com

	Elementor Pro
		Version: 3.19.3
		Author: Elementor.com

	Integration for Szamlazz.hu & WooCommerce
		Version: 5.9
		Author: Viszt Péter

	mame Saferpay for WooCommerce
		Version: 1.7.2
		Author: mame webdesign hüttig

	Rollbar
		Version: 2.7.1
		Author: Rollbar

	Sequential Order Numbers for WooCommerce
		Version: 1.10.1
		Author: SkyVerge

	ShortPixel Image Optimizer
		Version: 5.5.5
		Author: ShortPixel - Convert WebP/AVIF & Optimize Images

	UGL DHL Shipping
		Version: 1.0.0
		Author: UGL

	UGL Knowledgebase
		Version: 1.0.0
		Author: UGL

	WooCommerce
		Version: 8.6.1
		Author: Automattic

	WooCommerce Attribute Swatches by Iconic
		Version: 1.17.3
		Author: Iconic <support@iconicwp.com>

	WooCommerce EU VAT Number
		Version: 2.9.1
		Author: WooCommerce

	WooCommerce Nested Category Layout
		Version: 1.20.1
		Author: SkyVerge

	WooCommerce Product Configurator by Iconic
		Version: 1.21.2
		Author: Iconic

	WordSnippets
		Version: 1.0.0
		Author: László Monda

	WP Fastest Cache
		Version: 1.2.3
		Author: Emre Vona

	WP Fastest Cache Premium
		Version: 1.7.0
		Author: Emre Vona

	WP Mail SMTP Pro
		Version: 4.0.2
		Author: WP Mail SMTP

	WPS Hide Login
		Version: 1.9.13.2
		Author: WPServeur, NicolasKulka, wpformation


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	
	woocommerce: Active


== Elementor Experiments ==
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Inactive by default
	Grid Container: Inactive by default
	Default to New Theme Builder: Active by default
	Editor Top Bar: Inactive by default
	Optimized Gutenberg Loading: Active by default
	Build with AI: Active by default
	Landing Pages: Active by default
	Nested Elements: Active
	Lazy Load Background Images: Inactive by default
	Optimize Image Loading: Active by default
	Global Style Guide: Active
	Page Transitions: Inactive
	Notes: Inactive
	Display Conditions: Inactive by default
	Form Submissions: Inactive
	Scroll Snap: Inactive
	Menu: Active
	Taxonomy Filter: Inactive by default


== Log ==
	
Log: showing 20 of 372024-02-08 21:43:04 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.19.1',
  'to' => '3.19.2',
)]
2024-02-08 21:46:22 [info] elementor::elementor_updater Started 
2024-02-08 21:46:22 [info] Elementor/Upgrades - _on_each_version Start  
2024-02-08 21:46:22 [info] Elementor/Upgrades - _on_each_version Finished 
2024-02-08 21:46:22 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.19.1',
  'to' => '3.19.2',
)]
2024-02-27 14:09:20 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.19.2',
  'to' => '3.19.3',
)]
2024-02-27 14:09:21 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.19.2',
  'to' => '3.19.3',
)]
2024-02-27 14:09:21 [info] elementor-pro::elementor_pro_updater Started 
2024-02-27 14:09:21 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-02-27 14:09:21 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-02-27 14:09:21 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.19.2',
  'to' => '3.19.3',
)]
2024-02-27 14:09:21 [info] elementor::elementor_updater Started 
2024-02-27 14:09:21 [info] Elementor/Upgrades - _on_each_version Start  
2024-02-27 14:09:21 [info] Elementor/Upgrades - _on_each_version Finished 
2024-02-27 14:09:21 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.19.2',
  'to' => '3.19.3',
)]
2024-02-28 17:39:06 [info] elementor::elementor_updater Started 
2024-02-28 17:39:06 [info] Elementor/Upgrades - _on_each_version Start  
2024-02-28 17:39:06 [info] Elementor/Upgrades - _on_each_version Finished 
2024-02-28 17:39:06 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.19.3',
  'to' => '3.19.4',
)]
2024-02-28 17:39:06 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.19.3',
  'to' => '3.19.4',
)]

PHP: showing 5 of 5PHP: 2024-02-05 17:42:43 [warning X 1][../wp-content/plugins/elementor/core/files/manager.php::111] unlink(../wp-content/uploads/elementor/css/post-37899.css): No such file or directory [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-02-12 02:57:27 [error X 1][../wp-content/plugins/elementor/data/v2/base/controller.php::227] Uncaught Elementor\Data\V2\Base\Exceptions\WP_Error_Exception: Method 'WP_REST_Controller::get_item_permissions_check' not implemented. Must be overridden in subclass. in ../wp-content/plugins/elementor/data/v2/base/controller.php:227
Stack trace:
#0 ../wp-content/plugins/elementor/data/v2/base/base-route.php(105): Elementor\Data\V2\Base\Controller->get_permission_callback(Object(WP_REST_Request))
#1 ../wp-content/plugins/elementor/data/v2/base/base-route.php(242): Elementor\Data\V2\Base\Base_Route->get_permission_callback(Object(WP_REST_Request))
#2 ../wp-includes/rest-api/class-wp-rest-server.php(1160): Elementor\Data\V2\Base\Base_Route->Elementor\Data\V2\Base\{closure}(Object(WP_REST_Request))
#3 ../wp-includes/rest-api/class-wp-rest-server.php(1041): WP_REST_Server->respond_to_request(Object(WP_REST_Request), '...', Array, NULL)
#4 ../wp-includes/rest-api/class-wp-rest-server.php(431): WP_REST_Server->dispatch(Object(WP_REST_Request))
#5 ../wp-includes/rest-api.php(424): WP_REST_Server->serve_request('...')
#6 ../wp-includes/class-wp-hook.php(324): rest_api_loaded(Object(WP))
#7 ../wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#8 ../wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#9 ../wp-includes/class-wp.php(418): do_action_ref_array('...', Array)
#10 ../wp-includes/class-wp.php(813): WP->parse_request('')
#11 ../wp-includes/functions.php(1336): WP->main('')
#12 ../wp-blog-header.php(16): wp()
#13 ../index.php(17): require('...')
#14 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-02-13 18:46:18 [warning X 2][../wp-content/plugins/elementor/includes/elements/container.php::1845] Undefined array key "size" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-02-27 16:08:17 [notice X 102][../wp-content/plugins/elementor-pro/modules/woocommerce/module.php::627] parse_str(): Passing null to parameter #1 ($string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-03-06 22:49:14 [notice X 1][../wp-content/plugins/elementor/includes/base/controls-stack.php::228] hexdec(): Passing null to parameter #1 ($hex_string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatibility not specified

== Elementor Pro - Compatibility Tag ==

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by toineenzo** [comment]
Does this also happen when you disable your WooCommerce related plugins (except WooCommerce)? Also, looks like you're using Avada on top of Elementor?

**Comment by mondalaci** [comment]
The issue persists when disabling every plugin except WooCommerce, Elementor, and Elementor Pro.

I want to remove Avada, but some parts of our site still depend on it.

**Comment by ntentopoulos** [comment]
@mondalaci try to downgrade your PHP version on the server.
Same issue with PHP 8.1.27, downgraded to 8.0.30 and no more warning message, also checkout is working fine now.

**Comment by mondalaci** [comment]
@ntentopoulos Thanks, but I won't put in extra work to delay the inevitable transition to more recent PHP versions. I strongly believe that warnings like this should be fixed promptly. It wouldn't take much effort.

**Comment by nicholaszein** [comment]
Hello @mondalaci!

Thank you for your submission! 🙏 

### 🔗 🐞 This issue is a Duplicate of #24880 

Which is on the same subject and was solved with the release of Elementor Pro version 3.20.0.

Keep in mind that neither WordPress nor WooCommerce have full support for PHP 8.2+ yet, and it's not recommended to use it on production websites.

### 🔍 Please, in the future, remember to check if there isn't any thread similar to yours already opened. 

Kind regards

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.20.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
